semantics/language/execution/expr/assignment.k

```diff
@@ -39,7 +39,7 @@ module C-EXPR-ASSIGNMENT
           ~> checkArrayStaticAssign(T', V)
           ~> checkRestrictAssign(getRestrictBlock(T), getRelevantTags(Loc, V))
           ~> write(Loc, V, T')
-          ~> transferProv(Loc, tv(V, T))
+          ~> transferProv(Loc, tv(V, stripQualifiers(T)))
           requires isRestrictType(T)
```

Comment:
The changes above shouldn't be needed, I don't think, or else the strip should happen sooner (when the actual lvalue conversion occurs).